### PR TITLE
Support JetBrains IDEs 2024.1

### PR DESCRIPTION
**Pull request by tpasternak**
# Checklist

- [ ] I have filed an issue about this change and discussed potential changes with the maintainers.
- [ ] I have received the approval from the maintainers to make this change.
- [ ] This is not a stylistic, refactoring, or cleanup change.

Please note that the maintainers will not be reviewing this change until all checkboxes are ticked. See 
the [Contributions](https://github.com/bazelbuild/intellij#contributions) section in the README for more 
details.

# Discussion thread for this change

Issue number: `<please reference the issue number or url here>`

# Description of this change

most of the files are just copies, you can compare them with:
```
diff -r sdkcompat/v2{33,41}
diff -r cpp/sdkcompat/v2{33,41}
diff -r testing/testcompat/v2{33,41}
diff intellij_platform_sdk/BUILD.idea2{33,41}
diff intellij_platform_sdk/BUILD.ue2{33,41}
diff intellij_platform_sdk/BUILD.clion2{33,41}
```

### CODE REVIEW

**Comment by ahmedyarub** [comment]
> @ujohnny i'm getting this for clion, if something comes to your mind rn, then plase share, otherwise I'll do it myself
> 
> ```
> ERROR: clwb/BUILD:58:13: Building clwb/libclwb_lib.jar (31 source files) failed: (Exit 1): java failed: error executing Javac command (from target //clwb:clwb_lib) external/remotejdk17_linux_aarch64/bin/java '--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED' '--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED' ... (remaining 19 arguments skipped)
> clwb/src/com/google/idea/blaze/clwb/run/GoogleTestUtilAdapter.java:35: error: cannot find symbol
>     GoogleTestUtilAdapter adapter = EP_NAME.getPoint().extensions().findFirst().orElse(null);
>                                                       ^
>   symbol:   method extensions()
>   location: interface ExtensionPoint<GoogleTestUtilAdapter>
> Target //clwb:clwb_bazel_zip failed to build
> Use --verbose_failures to see the command lines of failed build steps.
> INFO: Elapsed time: 2.659s, Critical Path: 2.16s
> INFO: 4 processes: 4 internal.
> ERROR: Build did NOT complete successfully
> ```

Does this help?
https://github.com/ujohnny/intellij/commit/eb472e4a77668d5ba733925d49f18d182000fd45

File: common/util/BUILD [hunk 1]
```diff
@@ -42,6 +42,7 @@ java_library(
     deps = [
         "//intellij_platform_sdk:jsr305",  # unuseddeps: keep for @Nullable
         "//intellij_platform_sdk:plugin_api",
+	"@error_prone_annotations//jar",
     ],
```
Comment (ahmedyarub) [inline common/util/BUILD:46]: I think that there is a formatting problem in here. I think that you switched between tabs and spaces or something.

File: intellij_platform_sdk/BUILD [hunk 2]
```diff
@@ -601,7 +627,7 @@ config_setting(
 config_setting(
     name = "clion-2023.2-mac",
     values = {
-        "cpu": "darwin_x86_64", #todo how about m1?
+        "cpu": "darwin_x86_64",  #todo how about m1?
```
Comment (ujohnny) [inline intellij_platform_sdk/BUILD:630]: I'm not sure if you'd like to be in blame list for this line with this PR

File: sdkcompat/BUILD [hunk 3]
```diff
@@ -15,6 +15,9 @@ java_library(
     name = "sdkcompat",
     visibility = SDK_COMPAT_VISIBILITY,
     exports = select_for_plugin_api({
+        "clion-2024.1": ["//sdkcompat/v241"],
```
Comment (ujohnny) [inline sdkcompat/BUILD:18]: should we keep these sorted lexicographically?

File: third_party/go/BUILD [hunk 4]
```diff
@@ -22,6 +22,9 @@ java_library(
         "intellij-ue-2023.2": ["@go_2023_2//:go"],
         "intellij-2023.3": ["@go_2023_3//:go"],
         "intellij-ue-2023.3": ["@go_2023_3//:go"],
+        "intellij-2024.1": ["@go_2024_1//:go"],
+        "intellij-ue-2024.1": ["@go_2024_1//:go"],
+
```
Comment (ujohnny) [inline third_party/go/BUILD:27]: nit: whitespace

File: java/BUILD [hunk 5]
```diff
@@ -57,6 +57,7 @@ java_library(
         "//sdkcompat",
         "//shared",
         "//third_party/auto_value",
+        "@error_prone_annotations//jar"
```
Comment (ujohnny) [inline java/BUILD:60]: please add trailing comma

File: intellij_platform_sdk/build_defs.bzl [hunk 6]
```diff
@@ -104,6 +104,23 @@ def _check_channel_map():
         fail("Unexpected values in INDIRECT_PRODUCT_CHANNELS: %s" % unexpected)
 
 DIRECT_IJ_PRODUCTS = {
+    "clion-2024.1": struct(
```
Comment (ujohnny) [inline intellij_platform_sdk/build_defs.bzl:107]: same comment re lexicographic order

**Review by ujohnny — CHANGES_REQUESTED** [review]
some nits, also reached you privately regarding copyrights. otherwise lgtm

**Review by mai93 — COMMENTED** [review]
thanks @tpasternak. Left minor comments

**Review by ujohnny — APPROVED** [review]

File: intellij_platform_sdk/BUILD.clion241 [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+java_import(
+    name = "guava",
+    jars = glob([
+        "lib/lib.jar",
+        "lib/lib-client.jar",
+    ]),
+)
+
+java_library(name = "jsr305")  # not bundled
```
Comment (mai93) [inline intellij_platform_sdk/BUILD.clion241:50]: here it is marked as unmarked, so should we link it?

Comment (mai93) [reply intellij_platform_sdk/BUILD.clion241:50]: SGTM

**Review by mai93 — APPROVED** [review]

**Pull request merged**